### ISSUE TITLE
Unable to download nilmtk

### ISSUE THREAD

**Issue by xXLordVoldemord** [nilmtk/nilmtk#953]
Hey guys, I just tried to download nilmtk on several computers, both windows and apple. I wasnt able to download any of the versions (I have tried 0.4.1,0.4.2. and 0.4.3). I always get the same errors see appendix. I am somewhat new to all this, am I doing sth wrong? Thanks a lot for your help
![NILMTK](https://user-images.githubusercontent.com/38938930/167304039-19fbc5ca-459c-448c-a564-9950831438f8.PNG)
!

**Comment by TristanMenzinger** [comment]
I had the same issue and managed to work around this as follows:

From the [nilmtk repository on anaconda](https://anaconda.org/nilmtk/repo) download the`.tar.bz2` files for your architecture for both the [nilm metadata](https://anaconda.org/nilmtk/nilm_metadata/files) package and the [nilmtk](https://anaconda.org/nilmtk/nilmtk/files) package.

Then, run `conda install nilmtk-X.X.X-py_0.tar.bz2` and `conda install nilm_metadata-X.X.X.tar.bz2` with the X's as the version number of the package you want to install.

**Comment by camilomarino** [comment]
The solution from @TristanMenzinger  I think should work, however I usually install it another way and so far I have had no problems.

I recommend creating an environment with python 3.8 or lower and install via pip, the following line should work to install nilmtk:
``
pip install git+https://github.com/nilmtk/nilmtk@0.4.3 
``

**Comment by Rebecca29** [comment]
@camilomarino It really work smoothly by this way. Thx!

**Comment by aarif0203028** [comment]
> I had the same issue and managed to work around this as follows:
> 
> From the [nilmtk repository on anaconda](https://anaconda.org/nilmtk/repo) download the`.tar.bz2` files for your architecture for both the [nilm metadata](https://anaconda.org/nilmtk/nilm_metadata/files) package and the [nilmtk](https://anaconda.org/nilmtk/nilmtk/files) package.
> 
> Then, run `conda install nilmtk-X.X.X-py_0.tar.bz2` and `conda install nilm_metadata-X.X.X.tar.bz2` with the X's as the version number of the package you want to install.

@TristanMenzinger do we need to place `.tar.bz2` files in any specific folder?

**Comment by Rebecca29** [comment]
Have you tried to put it under the site-packages folder?

**Comment by TristanMenzinger** [comment]
> > I had the same issue and managed to work around this as follows:
> > From the [nilmtk repository on anaconda](https://anaconda.org/nilmtk/repo) download the`.tar.bz2` files for your architecture for both the [nilm metadata](https://anaconda.org/nilmtk/nilm_metadata/files) package and the [nilmtk](https://anaconda.org/nilmtk/nilmtk/files) package.
> > Then, run `conda install nilmtk-X.X.X-py_0.tar.bz2` and `conda install nilm_metadata-X.X.X.tar.bz2` with the X's as the version number of the package you want to install.
> 
> @TristanMenzinger do we need to place `.tar.bz2` files in any specific folder?

As @Rebecca29 said, anywhere is fine. You don't need to create an environment at all, using the default (= no environment) is fine. In the IDE of your choice (i.e. Jupyter Notebook) simply proceed with the commands. Can you try running the commands outside of an environment using pip (i.e. `pip install nilm_metadata-X.X.X.tar.bz2` and `pip install install nilmtk-X.X.X-py_0.tar.bz2` or try @camilomarino's solution.

Generally, I would highly recommend against using anaconda and use pip over conda whenever possible, but that's just me. Conda has certain advantages with binding-heavy packages, but this isn't the case here.

**Comment by aarif0203028** [comment]
I was able to make it work without creating an environment. Thank you @Rebecca29  and @TristanMenzinger for your inputs.

**Comment by aalaoui12** [comment]
The solution from @camilomarino  worked perfectly, but keep in mind that the Python version must be exactly 3.8

**Comment by Raphael164** [comment]
Thanks for the posted solutions. 
Now that i have installed nilmtk (hopefully working), I've stumbled across some errors concerning the DataSet part of nilmtk:

![image](https://user-images.githubusercontent.com/124141239/221150767-d7a10d25-0597-4f84-8a0d-436645ae8716.png)

![image](https://user-images.githubusercontent.com/124141239/221151884-762e1992-f9f7-4def-a26e-89c4bde98ecc.png)

**Comment by Raphael164** [comment]
Just in case anyone has the same problem. For me reinstalling the nilm_metadata with the newest version solved the problem (i am on windows):

`pip uninstall nilm_metadata
`
`pip install git+https://github.com/nilmtk/nilm_metadata`

**Comment by crn565** [comment]
I am trying to install nilmtk in Ubuntu for problems of memory in Windows . I tried the instructions of Rafphael164 and finally I can  load  the  dataset  .. but the code faults because  (I think)  is using python3.9.
Anyone know what version  of  python  must be works with nilmtk?

**Comment by Raphael164** [comment]
Are you able to downgrade your python version to 3.8 or lower? If so, it should solve your problems.

Otherwise I would suggest you this article, which explains the installation of NILMTK on Linux OS: [https://klemenjak.medium.com/a-step-by-step-manual-for-installing-nilmtk-bff86e3aa418](https://github.com/nilmtk/nilmtk/issues/url)

**Comment by Jehan98** [comment]
> The solution from @TristanMenzinger I think should work, however I usually install it another way and so far I have had no problems.
> 
> I recommend creating an environment with python 3.8 or lower and install via pip, the following line should work to install nilmtk: `pip install git+https://github.com/nilmtk/nilmtk@0.4.3 `

Tried but not working 

![image](https://github.com/nilmtk/nilmtk/assets/72525559/8d114757-1822-49c4-93a3-87056f90de8c)

**Comment by mucrolores** [comment]
@camilomarino's solution was immensely helpful in successfully installing the required components, and I'd like to express my gratitude.  

Here's some additional information for the latest version:
To ensure that Python can successfully import `nilmtk`,I will need to install the [nilm_metadata](https://github.com/nilmtk/nilm_metadata) package. This can be achieved by using the following command `pip install git+https://github.com/nilmtk/nilm_metadata@0.2.5` to make python runs `import nilmtk` successfully.

In summary, for the most up-to-date installation instructions, here're the steps
1. Create environment for python=3.8
2. `pip install git+https://github.com/nilmtk/nilmtk@0.4.3` Ref from [here](https://github.com/nilmtk/nilmtk/issues/953#issuecomment-1120559309) and [here](https://github.com/nilmtk/nilmtk/issues/953#issuecomment-1120473149)
3. `pip install git+https://github.com/nilmtk/nilm_metadata@0.2.5`
Verify installation
1. startup python console environment and enter `import nilmtk`

**Comment by chilin0525** [comment]
@mucrolores  You save my life.

**Comment by mixalako** [comment]
@mucrolores Hello. What pip version are you using?

**Comment by chilin0525** [comment]
@mixalako 
It works on my computer after testing on new environment. But you need to remove `0.2.5`.

**Comment by mucrolores** [comment]
Apologies, @mixalako. My mistake – I provided the incorrect version number for nilm_metadata. To resolve the installation issue, you can use the following command:
`pip install git+https://github.com/nilmtk/nilm_metadata@0.2.4`
And my pip version is `23.3`.

**Comment by mixalako** [comment]
@mucrolores Thank you very much for your clarifications they have been very helpful. Another issue is about the datasets, i cannot find neither REDD nor iAWE. Thanks again.

 For everyones information there is also [nilm_metadata](https://github.com/nilmtk/nilm_metadata) which holds a setup.py for those who cannot install the metadata with pip.

**Comment by mucrolores** [comment]
@mixalako, I regret to inform you that I couldn't locate those two datasets either. It seems that these datasets are no longer available, and I was unable to find any alternative links.

**Comment by RajatPatodi** [comment]
![error_nilmtk_installation](https://github.com/nilmtk/nilmtk/assets/94966634/6d6a4ffe-b76f-47ec-b0c5-86ce14d9ecb9)

@mucrolores  , @camilomarino   I followed the steps you guys had mentioned and imported nilmtk successfully (I guess), but it fails to import futher lined and I'm getting this error. 

any help!!

**Comment by mucrolores** [comment]
@RajatPatodi, it appears that you're using an outdated version of the example. The module nilmtk.cross_validation has been removed in the latest version of the library. Please refer to the [API documentation](https://nilmtk.github.io/nilmtk/master/index.html#) for the updated information.

**Comment by RajatPatodi** [comment]
> @RajatPatodi, it appears that you're using an outdated version of the example. The module nilmtk.cross_validation has been removed in the latest version of the library. Please refer to the [API documentation](https://nilmtk.github.io/nilmtk/master/index.html#) for the updated information.

THANKS, can you also suggest a way that I can learn using this toolkit for my project along with some examples
